### PR TITLE
Fix a crash when `__all__` exists but cannot be inferred

### LINKED ISSUE #8740: The undefined variable '__all__' results in an inference failed
The undefined variable runners uses an undefined attributes \_\_all\_\_, pylint reports an inference failed. See the following example.

pylint_example.py
```
import sys
if sys.version_info >= (3, 7):
    __all__ += runners.__all__
```

The expected behavior:
Pylint does not crash.

The actual output:
```
>> pylint pylint_example.py
************* Module pylint_example
Desktop/pylint_example.py:19:0: C0303: Trailing whitespace (trailing-whitespace)
Desktop/pylint_example.py:56:0: C0304: Final newline missing (missing-final-newline)
Desktop/pylint_example.py:1:0: C0114: Missing module docstring (missing-module-docstring)
Desktop/pylint_example.py:55:4: E0602: Undefined variable '__all__' (undefined-variable)
Desktop/pylint_example.py:55:15: E0602: Undefined variable 'runners' (undefined-variable)
Exception on node <Module.pylint_example l.0 at 0x7f797324d190> in file '/home/xxm/Desktop/pylint_example.py'
Traceback (most recent call last):
  File "/home/xxm/.local/lib/python3.11/site-packages/pylint/utils/ast_walker.py", line 96, in walk
    callback(astroid)
  File "/home/xxm/.local/lib/python3.11/site-packages/pylint/checkers/variables.py", line 1364, in leave_module
    self._check_all(node, not_consumed)
  File "/home/xxm/.local/lib/python3.11/site-packages/pylint/checkers/variables.py", line 3049, in _check_all
    assigned = next(node.igetattr("__all__"))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/xxm/.local/lib/python3.11/site-packages/astroid/bases.py", line 197, in _infer_stmts
    raise InferenceError(
astroid.exceptions.InferenceError: Inference failed for all members of [<AssignName.__all__ l.55 at 0x7f797324da10>].
Desktop/empirical_type_checker/bug_report/pylint_example.py:1:0: F0002: /home/xxm/Desktop/pylint_example.py: Fatal error while checking '/home/xxm/Desktop/pylint_example.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/home/xxm/.cache/pylint/pylint-crash-2023-06-07-00-29-01.txt'. (astroid-error)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

#### Test Environment:
Ubuntu 18.04,
CPython 3.11.3
Pylint: 3.0.0b1
astroid 3.0.0a3

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #8740

### CODE REVIEW

File: tests/functional/u/undefined/undefined_all_variable_edge_case.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+"""Edge case: __all__ exists in module's locals, but cannot be inferred.
+
+Other tests for undefined-all-variable in tests/functional/n/names_in__all__.py"""
+
+__all__ += []  # [undefined-variable]
```
Comment (nickdrozd) [inline tests/functional/u/undefined/undefined_all_variable_edge_case.py:5]: I'm not well-versed in using `__all__`. Is it the case that it doesn't exist until it's defined explicitly? (As opposed to being an always-defined global)

Comment (jacobtylerwalls) [reply tests/functional/u/undefined/undefined_all_variable_edge_case.py:5]: I probably shouldn't have used the word "exists".

When __all__ isn't defined, you lose the ability to do:
```python
>>> from pylint import *
>>> version
'3.0.0b1'
```

> Is it the case that it doesn't exist until it's defined explicitly? (As opposed to being an always-defined global)

Indeed!
```python
>>> import astroid
>>> astroid.__all__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'astroid' has no attribute '__all__'. Did you mean: '__file__'?
```

Comment (jacobtylerwalls) [reply tests/functional/u/undefined/undefined_all_variable_edge_case.py:5]: Versus
```python
>>> pylint.__all__
['__version__', 'version', 'modify_sys_path', 'run_pylint', 'run_symilar', 'run_pyreverse']
```

File: pylint/checkers/variables.py [hunk 2]
```diff
@@ -3046,7 +3046,10 @@ def _check_module_attrs(
     def _check_all(
         self, node: nodes.Module, not_consumed: dict[str, list[nodes.NodeNG]]
     ) -> None:
-        assigned = next(node.igetattr("__all__"))
+        try:
+            assigned = next(node.igetattr("__all__"))
```
Comment (DanielNoord) [inline pylint/checkers/variables.py:3050]: Why don't we use `safe_infer` here?

Comment (jacobtylerwalls) [reply pylint/checkers/variables.py:3050]: Do you know of a way to use a helper with `igetattr` (or to use something besides igetattr, here)?

Comment (DanielNoord) [reply pylint/checkers/variables.py:3050]: Ah, of course.. Well, seems like we need `safe_igetattr` 😄

**Review by DanielNoord — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you @jacobtylerwalls. Just wondering if an internal checker for unsafe inference wouldn't be more practical than typing astroid properly 😄
